.github/workflows/codeql.yml

```diff
@@ -0,0 +1,25 @@
+name: CodeQL Analysis
+
```

Comment:
zizmor will probably complain about not having `permissions: {}` here -- worth running as an improvement